src/Nncase.Core/Diagnostics/DumpScope.cs

```diff
@@ -55,8 +55,21 @@ public DumpScope(IDumpper newDumpper, IServiceProvider? serviceProvider = null)
     /// </summary>
     /// <param name="serviceProvider">Service provider.</param>
     /// <returns>Current dumpper.</returns>
-    public static IDumpper GetCurrent(IServiceProvider? serviceProvider) =>
-        _dumpper.Value ??= (serviceProvider ?? CompileSessionScope.Current)?.GetRequiredService<IDumpperFactory>().Root ?? NullDumpper.Instance;
+    public static IDumpper GetCurrent(IServiceProvider? serviceProvider)
+    {
+        if (_dumpper.Value is null)
+        {
+            var provider = serviceProvider ?? CompileSessionScope.Current;
+            if (provider is null)
+            {
+                return NullDumpper.Instance;
+            }
+
+            _dumpper.Value = provider.GetRequiredService<IDumpperFactory>().Root;
```

Comment:
When there is no dump scope, it's not proper to store dumppers.

Reply:
fixed